Update service broker descriptions

adborden opened the pull request:
Part of https://favro.com/card/1e11108a2da81e3bd7153a7a/18F-989

Let's make sure the we're on the same page about the service descriptions in the docs before updating them in the brokers. Once we have them set, I can copy them out to the individual broker repos.

These descriptions are based on the broker descriptions plus edits from @brittag.

adborden committed:
Update service broker descriptions

Commit:
Merge branch 'master' into ab/update-service-broker-descriptions

brittag commented on content/docs/services/elasticsearch17.md:
```diff
@@ -4,7 +4,7 @@ menu:
 title: Elasticsearch 1.7
 name: "elasticsearch17"
-description: "[Elasticsearch](https://www.elastic.co/) version 1.7"
+description: "[Elasticsearch](https://www.elastic.co/products/elasticsearch) version 1.7 a distributed, RESTful search and analytics engine"
```
This needs something like a colon after the version number, and I suspect links aren't a good idea since they wouldn't show up properly in the CLI - `Elasticsearch version 2.3: a distributed, RESTful search and analytics engine`

brittag replied:
(Similar for the other Elasticsearch and Redis descriptions)

adborden replied:
Right, the links will be stripped out for the actual service broker metadata. The links here will only be displayed on the docs site. I think we should keep them since they are useful in an HTML context.

adborden replied:
BUT, these descriptions are actually for the page metadata (for things like url previews on slack) and are not converted to HTML so should not include links. I'll update that on all the pages.

brittag replied:
Cool, that's ok with me!

brittag commented on content/docs/services/cdn-route.md:
```diff
@@ -4,7 +4,7 @@ menu:
 title: CDN service
 name: "cdn-route"
-description: "Custom routes with HTTPS certificates using Amazon CloudFront"
+description: "Custom domains, CDN caching, and TLS certificates using Amazon CloudFront"
```
This should be `Custom domains, CDN caching using Amazon CloudFront, and TLS certificates` since the domains and TLS certificates aren't using CloudFront

brittag replied:
Actually, just `Custom domains, CDN caching, and TLS certificates with automatic renewal` would be fine here - we can leave the CDN details to the documentation

brittag commented on content/docs/services/cdn-route.md:
```diff
@@ -18,7 +18,7 @@ This service provides three key elements to support production applications:
 Plan Name | Description | Price
 --------- | ----------- | -----
-`cdn-route` | A CDN distribution with custom domain and free TLS certificate with auto renewal | Free
+`cdn-route` | Custom domains, CDN caching, and TLS certificates with auto renewal | Free
```
`Custom domains, CDN caching, and TLS certificates with automatic renewal`

brittag commented on content/docs/services/relational-database.md:
```diff
@@ -13,16 +14,16 @@ If your application uses relational databases for storage, you can use the AWS R
+`shared-psql`            | Shared infrastructure for PostgreSQL DB               | Free
+`medium-psql`            | Dedicated medium RDS PostgreSQL DB instance           | $0.115 / hr + storage
+`medium-psql-redundant`  | Dedicated redundant medium RDS PostgreSQL DB instance | $0.230 / hr + storage
+`large-psql`             | Dedicated large RDS PostgreSQL DB Instance            | $0.230 / hr + storage
```
`instance` instead of `Instance`

brittag commented on content/docs/services/relational-database.md:
```diff
@@ -13,16 +14,16 @@ If your application uses relational databases for storage, you can use the AWS R
-`medium-psql-redundant`  | Dedicated Redundant Medium RDS Postgres DB Instance | $0.230 / hr + storage
-`large-psql`             | Dedicated Large RDS Postgres DB Instance            | $0.230 / hr + storage
-`large-psql-redundant`   | Dedicated Redundant Large RDS Postgres DB Instance  | $0.470 / hr + storage
+`shared-psql`            | Shared infrastructure for PostgreSQL DB               | Free
```
Let's spell out "database" instead of "DB" in all of these descriptions. But specifically for this one, `Shared infrastructure for PostgreSQL DB` is a bit vague - what is it really? Is it really `Shared PostgresSQL database for prototyping (no sensitive or production data)` ?

adborden replied:
> Shared PostgresSQL database for prototyping (no sensitive or production data)

Yeah, good call.

brittag commented on content/docs/services/s3.md:
```diff
@@ -4,7 +4,7 @@ menu:
 title: S3
 name: "s3"
-description: "Amazon Simple Storage System (persistent storage)"
+description: "Amazon S3 provides developers with secure, durable, highly-scalable object storage"
```
Can we make this one match the RDS one better?

brittag replied:
More like `Secure, durable, highly-scalable object storage using Amazon S3`?

brittag commented on data/services/cdn-route.toml:
```diff
@@ -1,3 +1,3 @@
 name = "cdn-route"
-description = "Custom routes with HTTPS certificates using Amazon CloudFront"
+description = "Custom domains, CDN caching, and TLS certificates using Amazon CloudFront"
```
This should match whatever we change above for CDN routes. Should we have to update both manually, or is there a way in Hugo to use this data inside pages?

brittag replied:
Same comment/question for the other `.toml` items (They should match, and can we do it automatically?)

adborden replied:
The `.toml` files are what is used in the HTML table/pages and can be HTML/markdown content. The `description` in the markdown files are actually page metadata and should **not** contain HTML/markdown content.

The content _could_ be shared in jekyll, so I suspect there's a way to do it in hugo (although I'm not familiar with it). However, given that these fields contain different content types, I think it's better to keep them separate even if it means duplication.

brittag replied:
Ah interesting! That seems ok to me.

brittag replied:
Might be good to add a note about this bit in the cg-site README, so that we remember to update both next time we go through this process - up to you.

brittag commented on content/docs/services/relational-database.md:
```diff
@@ -13,16 +14,16 @@ If your application uses relational databases for storage, you can use the AWS R
+`medium-psql-redundant`  | Dedicated redundant medium RDS PostgreSQL DB instance | $0.230 / hr + storage
+`large-psql`             | Dedicated large RDS PostgreSQL DB Instance            | $0.230 / hr + storage
+`large-psql-redundant`   | Dedicated redundant large RDS PostgreSQL DB instance  | $0.470 / hr + storage
 `shared-mysql`           | Shared infrastructure for MySQL DB                  | Free
```
Same comment as for PostgresSQL - `Shared infrastructure for MySQL DB` is a bit vague - what is it really? Is it really `Shared MySQL database for prototyping (no sensitive or production data)` ?

brittag commented on content/docs/services/redis28.md:
```diff
@@ -14,7 +14,7 @@ cloud.gov offers [Redis](https://www.redis.io/) 2.8 as a service. **This service
 Plan Name | Description | Price
 --------- | ----------- | -----
-`standard` | Redis instance with unlimited storage | Free in Alpha
+`standard` | Redis instance with several GB storage | Free in Alpha
```
Is there any way we can make this more specific? How many GB?

adborden replied:
[10 GB](https://gsa-tts.slack.com/archives/cloud-gov-atlas/p1487111719009649) but this number was not selected with any meaningful testing or planning, so while it's in Alpha we want to keep it vague.

brittag replied:
Ah ha! I'd suggest explaining that on https://cloud.gov/docs/services/redis28/ - that it's currently 10 GB but may change.

brittag commented on content/docs/services/cloud-gov-service-account.md:
```diff
@@ -4,7 +4,7 @@ menu:
 title: cloud.gov service account
 name: "cloud-gov-service-account"
-description: "Manage cloud.gov service accounts with access to your organization"
+description: "cloud.gov service accounts with access to your organization"
```
Thinking that "cloud.gov service accounts with access to your organization" doesn't explain what the service account is _for_ - instead of talking about access to your organization, could this be something that defines what a service account is, such as "cloud.gov service accounts for automated access by programs"?

adborden replied:
Yeah, I struggled with this because we only have a single plan, and it's kind of open ended what these could be used for, or what plans might appear in the future. I like your suggestion.

brittag commented on content/docs/services/cloud-gov-service-account.md:
```diff
@@ -16,7 +16,7 @@ To set up your application to be deployed with an automated system, you need a d
 Plan Name | Description | Price
 --------- | ----------- | -----
-`space-deployer` | Provision one account with "SpaceDeveloper" access to the space where the service is created | Free
+`space-deployer` | An account for continuous deployment, limited to a single space | Free
```
Suggest "A service account for continuous deployment, limited to a single space" to emphasize that this is a service account instead of a human account

brittag commented on content/docs/services/cloud-gov-identity-provider.md:
```diff
@@ -4,7 +4,7 @@ menu:
 title: cloud.gov identity provider
 name: "cloud-gov-identity-provider"
-description: "Manage client credentials for authenticating cloud.gov users in your app"
+description: "Authenticate cloud.gov users in your app"
```
I wonder if we can find a more plain-language way of saying this - not sure if I'd understand "Authenticate cloud.gov users in your app" at first read if I didn't know what this service was. Maybe "Use cloud.gov authentication in your application as a login system for your users"?

jmcarp replied:
We already had a meeting specifically to nail down this text. I'm wondering how much more time this is worth--are we making this clearer or talking in circles?

brittag replied:
Hmm, it is a good question, thanks for asking. I'm honestly a bit lost in the various sources of info here - I'm not sure what the already-nailed-down text is (whether it's this text or other text somewhere else).

Overall I'm totally fine with merging the existing text in this PR - all of my comments should be considered enhancements and shouldn't be considered blockers.

@adborden suggested the PR would benefit from a bit of additional review to prevent future cycles of laborious metadata merges - happy to leave it up to him to decide whether to do more copy work or merge as-is. :)

adborden replied:
> Authenticate cloud.gov users in your app

This came from me. The thing that stuck out for me with the "Manage client credentials for authenticating cloud.gov users in your app" copy was the "manage" part. The service does no managing of credentials. It only provides you with client id/secret so that you can "Use cloud.gov authentication in your application as a login system for your users". Is that right @jmcarp?

Anyway, I'm happy to go ahead with all the suggested changes above. I don't mean to steamroll any previous thought and discussion for the identity-provider copy, but I would like another round to clarify the thinking behind it and maybe iterate on the existing text.

adborden replied:
Hmm, although given an oauth token, potentially you could do much more than just use cloud.gov as a login. You could access any cloud.gov resources on behalf of the user. Do we have examples of what users are building with this service?

brittag replied:
I believe Tock currently uses this service - to log into Tock, you go through the cloud.gov auth flow. Not sure what else / what the other capabilities are.

brittag replied:
Actually, I'm not sure whether Tock is using this or something else - I'll leave this question to somebody else.

adborden committed:
rfc

- page metadata: add colon
- page metadata: remove markdown, which won't be rendered properly
- cdn-route copy
- aws-rds copy
- service accounts copy

adborden committed:
Add uaa-auth to service list
